pkg/cluster/fetch.go

```
@@ -212,6 +227,14 @@ func BuildSpec(ctx context.Context, client Client, cluster *v1alpha1.Cluster) (*
 // BuildSpecFromConfig constructs a cluster.Spec for an eks-a cluster config by retrieving all dependencies objects from the cluster using a kubernetes client.
 func BuildSpecFromConfig(ctx context.Context, client Client, config *Config) (*Spec, error) {
 	bundlesName, bundlesNamespace := bundlesNamespacedKey(config.Cluster)
+	eksaRelease := &v1alpha1.EKSARelease{}
+	if config.Cluster.Spec.EksaVersion != nil {
```

we should throw an error if eksa version is nil when we get here